src/object/cli_shard.c

```diff
@@ -783,12 +784,27 @@ dc_rw_cb(tse_task_t *task, void *arg)
 		 * If any failure happens inside Cart, let's reset failure to
 		 * TIMEDOUT, so the upper layer can retry.
 		 */
-		D_ERROR(DF_UOID" (%s) RPC %d to %d/%d, flags %lx/%x, task %p failed, %s: "DF_RC"\n",
+		D_ERROR(DF_UOID " (%s) RPC %d to %d/%d, flags %lx/%x, task %p failed, iod_nr:%u "
+				"%s(%lu): " DF_RC "\n",
 			DP_UOID(orw->orw_oid), is_ec_obj ? "EC" : "non-EC", opc,
 			rw_args->rpc->cr_ep.ep_rank, rw_args->rpc->cr_ep.ep_tag,
 			(unsigned long)orw->orw_api_flags, orw->orw_flags, task,
-			orw->orw_bulks.ca_arrays != NULL ||
-			orw->orw_bulks.ca_count != 0 ? "DMA" : "non-DMA", DP_RC(ret));
+			orw->orw_iod_array.oia_iod_nr,
+			orw->orw_bulks.ca_arrays != NULL || orw->orw_bulks.ca_count != 0
+			    ? "DMA"
+			    : "non-DMA",
+			orw->orw_bulks.ca_count, DP_RC(ret));
```

Comment:
It seems only print `orw->orw_bulks.ca_count` is enough, do not need former `orw->orw_bulks.ca_arrays != NULL || orw->orw_bulks.ca_count != 0 ? "DMA": "non-DMA"`.

Reply:
I'd leave it as-is, since it could help to discover potential defect (like ca_arrays != NULL but ca_count == 0)